### PR TITLE
ci: expand build matrix

### PR DESCRIPTION
**Pull request by bcressey**
**Issue number:**
N/A

**Description of changes:**
Adds coverage for aarch64 and both Kubernetes variants.

**Testing done:**
Tested via PR.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by tjkirch — COMMENTED** [review]
It looks like the right change, but the question is whether we want to merge it before fixing the build issue causing flaky failures unrelated to our changes; we'd be four times as likely to show a useless error...

**Comment by bcressey** [comment]
> It looks like the right change, but the question is whether we want to merge it before fixing the build issue causing flaky failures unrelated to our changes; we'd be four times as likely to show a useless error...

+1

This also doubling as a stress test to help me chase down the problem. I had a faint hope that the AMI refresh for the builders would help, but no luck.

**Review by tjkirch — APPROVED** [review]
If controlling the job count helps with our failures, as it seems to, I'm happy!

**Review by tjkirch — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -5,8 +5,13 @@ on:
 jobs:
   build:
     runs-on: [self-hosted, linux, x64]
+    strategy:
+      matrix:
+        variant: [aws-k8s-1.15, aws-k8s-1.16]
```
Comment (jahkeup) [inline .github/workflows/build.yml:11]: Any reason, other than total run time, to not include `aws-dev` in this list?

File: Makefile.toml [hunk 2]
```diff
@@ -12,28 +12,33 @@ BUILDSYS_TIMESTAMP = { script = ["date +%s"] }
...
 BUILDSYS_UPSTREAM_SOURCE_FALLBACK = "false"
+
 # We require license checks to pass to build an image.  If you're working on a
 # local change and don't have license information yet, you can run with `-e
 # BUILDSYS_ALLOW_FAILED_LICENSE_CHECK=true` to allow the build to continue even
 # if the license check fails.
 BUILDSYS_ALLOW_FAILED_LICENSE_CHECK = "false"
 
+# This controls how many `docker build` commands we'll invoke at once.
+BUILDSYS_JOBS = "8"
```
Comment (jahkeup) [inline Makefile.toml:27]: Is this number respected in downstream builds (eg: `make -j1` for each respective process)?

File: .github/workflows/build.yml (hunk 1 above)
Comment (bcressey) [reply .github/workflows/build.yml:11]: I'm somewhat reluctant to promote it as a "Tier 1" target which is how I view this list. All of the packages it needs still get built, so we'd catch docker build failures for example.

File: Makefile.toml (hunk 2 above)
Comment (bcressey) [reply Makefile.toml:27]: It's not - they are somewhat orthogonal concepts. The kernel build can still make use of all cores and we want to keep that property even if we only have a few builds going at once.

Comment (jahkeup) [reply Makefile.toml:27]: I see. I read this as though the parameter were prefixed with `BUILDSYS_` where `BUILDSYS_JOBS` meant `JOBS` (as in `-j`) which I'm used to seeing respected up and down the build "call stack".

File: .github/workflows/build.yml (hunk 1 above)
Comment (jahkeup) [reply .github/workflows/build.yml:11]: Gotcha. I think we should still be building the variant as much as possible - perhaps we could include it in a non-blocking/non-critical way where it isn't required for PR to pass?

**Review by jahkeup — APPROVED** [review]

**Comment by jahkeup** [comment]
Darn - I clicked re-run wanting only the failed `aws-k8s-1.16, aarch64` build to be re-attempted. Sorry about the pending builds!

File: tools/buildsys/src/builder.rs [hunk 3]
```diff
@@ -140,7 +159,21 @@ fn build(target: &str, build_args: &str, tag: &str, output: &str) -> Result<()>
     let _ = docker(&rmi);
 
     // Build the image, which builds the artifacts we want.
-    docker(&build)?;
+    // HACK: work around a transient, known failure case with Docker.
```
Comment (tjkirch) [inline tools/buildsys/src/builder.rs]: Is it worth adding this capability to the `docker` function?  I could imagine a signature like:

```
fn docker(args: &[String], retry: Retry) {...}

enum Retry {
   No,
   Yes {
      count: NonZeroU8,
      retry_failure_messages: &[&str],
   }
}
```

**Review by tjkirch — APPROVED** [review]

File: tools/buildsys/src/builder.rs (hunk 3 above)
Comment (bcressey) [reply tools/buildsys/src/builder.rs]: That would be neat, not sure if it's worth it right now. Maybe when we have more than one known error?

Comment (tjkirch) [reply tools/buildsys/src/builder.rs]: I think my concern was more around the repeated code and the need for a "hack" than the list of strings specifically, but obviously I don't think it's a blocker.  It'll be nice to get around this issue.

**Review by tjkirch — APPROVED** [review]

**Review by jahkeup — APPROVED** [review]

**Pull request merged**